### ISSUE TITLE
Quasar cannot find exported modules if using script setup, typescript and pug

### ISSUE THREAD

**Issue by QozbroQqn** [quasarframework/quasar#16129]
### What happened?

Sidenote: Your Reporting an Issue | Contribution Guide is a 404

If using script setup, typescript and pug, quasar app fails to export interfaces/types:

![image](https://github.com/quasarframework/quasar/assets/125742475/f1b57820-aa45-4b23-a077-4ff28858cfdd)

I created the app with `yarn create quasar`:
- quasar cli
- typescript
- sass
- script setup
- no eslint or others

and just changed:
- package.json (added pug 3)
- src/components/EssentialLink.vue
- src/layouts/MainLayout.vue

It currently reports only the EssentialLinkProps but it also cannot find the interfaces exported in `src/component/models.ts`

### What did you expect to happen?

It finds exported interfaces/types

### Reproduction URL

https://stackblitz.com/edit/stackblitz-starters-vetrn3

### How to reproduce?

- cd into quasar-project
- yarn
- quasar dev
- look into browser console

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite), TypeScript Support

### Quasar info output

```Shell
Operating System - Linux(5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36 Edg/115.0.1901.188) - linux/x64
NodeJs - 16.20.0

Global packages
  NPM - 9.4.2
  yarn - 1.22.19
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.12.3 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.4.3 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.16.5 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.3.4 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.2.4
  pinia - Not installed
  vuex - Not installed
  vite - 2.9.16 -- Native-ESM powered web dev build tool
  eslint - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - stackblitzstartersvetrn3-nlqe
  en0 - 192.168.1.104
```

**Comment by dongwa** [comment]
No one, including the documentation, indicates that you can export variables or types from a .vue file. However, I have found some solutions to address this issue:

Using `import type { EssentialLinkProps } from 'components/EssentialLink.vue';`

![image](https://github.com/quasarframework/quasar/assets/38457491/18b3c268-47f9-41a3-b8d8-1c4d736e8b51)

Another way is that you can define your type in the 'src/components/models.ts' file.

In summary, there are numerous ways to address this matter, it shouldn't be considered an issue to Quasar.

**Comment by QozbroQqn** [comment]
You are wrong. I ofcourse can export types and interfaces in .vue files and I do this for years now!
In fact Quasar does this too because the exports are coming from the `yarn create quasar` provided by Quasar.

As I wrote I did a `yarn create quasar` and added pug. Nothing else. If you do the same without pug, the errors are gone.

**Comment by rstoenescu** [comment]
This is a Vue SFC issue, not Quasar.

**Comment by QozbroQqn** [comment]
> This is a Vue SFC issue, not Quasar.

It's not. A Nuxt 3 minimal starter with pug works as expected even with SFC.

**Edit:**
Just tested a plain vue 3 with SFC and it does work with pug too

**Comment by QozbroQqn** [comment]
Another strange thing (only in Quasar):
If I remove pug from <template> and rebuild (npm/yarn run dev) the app. Everything is working fine. If I then add pug to <template> again and wait until the HMR applies the changes, pug/quasar works as expected. After a rebuild (npm/yarn run dev) the errors reported in my first post are back.
